src/EmbedIndex/Program.cs

```diff
@@ -164,33 +165,86 @@ public static IEnumerable<string> ComputeIndexEntries(string archiveRelativePath
+
         private static IEnumerable<Tuple<string, string>> ComputeCatalogEntries(ZipArchive archive, IEnumerable<IFileFormatIndexer> indexers)
         {
-            foreach (ZipArchiveEntry catalogEntry in archive.Entries)
+            // check to make sure we have the files we need to do catalog indexing
+            ZipArchiveEntry catalogEntry = archive.GetEntry(SymbolCatalogFileName);
+            ZipArchiveEntry catalogedFilesEntry = archive.GetEntry(CatalogedSymbolListFileName);
+
+            if (catalogEntry == null)
+            {
+                yield break;
+            }
+
+            if (catalogedFilesEntry == null)
             {
-                if (!catalogEntry.Name.ToLowerInvariant().EndsWith(".cat"))
+                yield break;
+            }
+
+            // create catalog_index_id.txt with SHA-256 of catalog
+            // store in root
+            string catalogHash;
+            using (Stream catalogStream = catalogEntry.Open())
+            {
+                using (HashAlgorithm sha = SHA256.Create())
                 {
-                    continue;
+                    catalogHash = sha.ComputeHash(catalogStream).ToHexString();
+                    ZipArchiveEntry catalogIndexEntry = archive.CreateEntry(CatalogIndexFileName);
+                    using (Stream indexStream = catalogIndexEntry.Open())
+                    {
+                        using (TextWriter writer = new StreamWriter(indexStream))
+                        {
+                            writer.WriteLine(catalogHash);
+                            writer.Flush();
+                        }
+                    }
                 }
-                string signedFile = catalogEntry.FullName.Substring(0, catalogEntry.FullName.Length - 4);
-                ZipArchiveEntry signedFileEntry = archive.Entries.SingleOrDefault(e => e.FullName == signedFile);
-                if (signedFileEntry == null)
+            }
+
+            // get list of cataloged symbol files
+            string[] catalogedFiles;
+            using (Stream catalogedFilesStream = catalogedFilesEntry.Open())
+            {
+                using (StreamReader reader = new StreamReader(catalogedFilesStream))
                 {
-                    continue;
+                    string content = reader.ReadToEnd();
+                    catalogedFiles = content.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
                 }
+            }
+            catalogedFilesEntry.Delete();
+
+            // for each regular symbol index entry covered by a symbol catalog,
+            // add an entry for the catalog file
+            HashSet<Tuple<string, string>> seen = new HashSet<Tuple<string, string>>();
+            foreach (string catalogedFile in catalogedFiles)
+            {
+                ZipArchiveEntry fileEntry = archive.GetEntry(catalogedFile);
                 foreach (IFileFormatIndexer indexer in indexers)
```

Comment:
Have you double checked that we only wind up generating one catalog index per file? Its not huge deal if we generated other ones by iterating all indexers (such as the SHA-1 source index), but it would be a little wasteful of space.